lib/block_view_lockups.go

```diff
@@ -818,109 +1174,541 @@ func (bav *UtxoView) _connectCoinLockup(
+				if proposedLockedBalanceEntry.VestingEndTimestampNanoSecs >
+					existingLockedBalanceEntry.VestingEndTimestampNanoSecs {
+					// NOTE: This case is particularly interesting as there's two situations in
+					// which we might find ourselves.
+					//
+					// First case:
+					// proposed:   <----------------------------------------->
+					// existing:   <------------------>
+					//                                 ^                     ^ Overhang
+					// Second case:
+					// proposed:   <----------------------------------------->
+					// existing:   <------------------>            <--------->
+					//                                 ^          ^ Overhang
+					// In the second case there exists another conflicting LockedBalanceEntry
+					// sometime in the future that we must be aware of. Also note that
+					// the first case is only possible in the very last iteration.
+					//
+					// To account for this, we split from the proposed LockedBalanceEntry
+					// and combine the left overlapping portion. This leaves us with two
+					// remaining cases:
+					// First case:
+					// proposed:                       <--------------------->
+					// existing:
+					//
+					// Second case:
+					// proposed:                       <--------------------->
+					// existing:                                   <--------->
+					//
+					// The second case is fine to leave as it will be taken care of in the
+					// subsequent iteration. However, we make a special note to capture the
+					// remaining LockedBalanceEntry present in the first case.
+
+					// We check if there's another locked balance entry sometime in the future.
+					splitLockedBalanceEntry, remainingLockedBalanceEntry, err := SplitVestedLockedBalanceEntry(
+						proposedLockedBalanceEntry,
+						existingLockedBalanceEntry.UnlockTimestampNanoSecs,
+						existingLockedBalanceEntry.VestingEndTimestampNanoSecs)
+					if err != nil {
+						return 0, 0, nil,
+							errors.Wrap(err, "_connectCoinLockup failed to compute vested split")
+					}
+
+					// Consolidate the split and existing locked balance entry.
+					combinedBalanceBaseUnits, err := SafeUint256().Add(
+						&splitLockedBalanceEntry.BalanceBaseUnits,
+						&existingLockedBalanceEntry.BalanceBaseUnits)
+					if err != nil {
+						return 0, 0, nil,
+							errors.Wrap(RuleErrorCoinLockupYieldCausesOverflowInLockedBalanceEntry,
+								"_connectCoinLockup")
+					}
+					splitLockedBalanceEntry.BalanceBaseUnits = *combinedBalanceBaseUnits
+
+					// Set the now combined splitLockedBalanceEntry into the view.
+					bav._setLockedBalanceEntry(splitLockedBalanceEntry)
+
+					// Update the proposed locked balance entry with the remaining portion.
+					proposedLockedBalanceEntry = remainingLockedBalanceEntry
+
+					// (3b-iii) On the final iteration, the remaining proposedLockedBalanceEntry
+					//          is the only vesting schedule left.
+					if ii == len(lockedBalanceEntries)-1 {
```

Comment:
I may be missing something but I *think* there's a bug here. I notice that you consolidate the proposed with the existing in this case, but you don't update the existing. So imagine a simple case like this:
```
proposed: <==================================>
existing: <============>
```

After this consolidation, I think you WANT to have:
```
proposed:               <=====================>
existing: <============>
```

And I think the timestamps are correct for the existing piece but NOT the balance. The balance of the existing seems like it will be whatever you had prior to consolidation, since you don't update the existing to be equal to the splitLockedBalanceEntry.

Lmk if I'm missing something here.

Reply:
I see... I think it doesn't matter because you won't use the existingBalanceEntry for anything after this point because they don't overlap. Interesting.